src/node_jsrtapi.cc

```diff
@@ -364,69 +403,78 @@ napi_status napi_set_property(napi_env e, napi_value o,
   return napi_ok;
 }
 
-napi_status napi_define_property(napi_env e, napi_value o,
-    napi_property_descriptor* p) {
-  napi_value descriptor;
-  CHECK_NAPI(napi_create_object(e, &descriptor));
-
+napi_status napi_define_properties(napi_env e,
+                                   napi_value o,
+                                   int property_count,
+                                   const napi_property_descriptor* properties) {
   napi_propertyname configurableProperty;
   CHECK_NAPI(napi_property_name(e, "configurable", &configurableProperty));
```

Comment:
> I wonder if this should be napi_get_property_name, to be consistent with other NAPI APIs.

Actually @boingoing and I have discussed eliminating the `napi_propertyname` type, because it seems to be a JSRT concept that leaked into the NAPI public API. V8 allows using any value type as a property key, and apparently leveldown makes use of that.

> this method calls other NAPI APIs- is there a reason for that vs. straight up calling JSRT APIs

Just for convenience & maintainability. Why not?